netbench-orchestrator/src/russula/netbench/client_worker.rs

```diff
@@ -0,0 +1,245 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::ClientContext;
+use crate::russula::states::StateApi;
+use crate::russula::states::TransitionStep;
+use crate::russula::{
+    error::{RussulaError, RussulaResult},
+    event::EventRecorder,
+    netbench::client::CoordState,
+    network_utils::Msg,
+    workflow::WorkflowTrait,
+};
+use core::fmt::Debug;
+use serde::{Deserialize, Serialize};
+use std::{fs::File, net::SocketAddr, process::Command};
+use sysinfo::{Pid, PidExt, ProcessExt, SystemExt};
+use tokio::net::{TcpListener, TcpStream};
+use tracing::{debug, info, warn};
+
+// Only used when creating a state variant for comparison
+const PLACEHOLDER_PID: u32 = 1000;
+
+/// Workflow state machine
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum WorkerState {
+    WaitCoordInit,
+    Ready,
+    Run,
+    Running(#[serde(skip)] u32),
+    RunningAwaitComplete(#[serde(skip)] u32),
+    Stopped,
+    Done,
+}
+
+/// Worker protocol for the client
+#[derive(Clone)]
+pub struct WorkerWorkflow {
+    id: String,
+    state: WorkerState,
+    peer_state: CoordState,
+    netbench_ctx: ClientContext,
+    event_recorder: EventRecorder,
+}
+
+impl WorkerWorkflow {
+    pub fn new(id: String, netbench_ctx: ClientContext) -> Self {
+        WorkerWorkflow {
+            id,
+            state: WorkerState::WaitCoordInit,
+            peer_state: CoordState::CheckWorker,
+            netbench_ctx,
+            event_recorder: EventRecorder::default(),
+        }
+    }
+}
+
+impl WorkflowTrait for WorkerWorkflow {
```

Comment:
Nit: `WorkflowTrait` feels a bit redundant, but might be too late to change that?

Reply:
Yea, I want to avoid nit refactors